language/move-vm/runtime/src/loader.rs

```diff
@@ -1875,3 +1877,45 @@ impl Loader {
         self.type_to_type_layout_impl(ty, 1)
     }
 }
+
+// Public APIs for external uses.
+impl Loader {
+    pub fn get_function_signature(
+        &self,
+        function_name: &IdentStr,
+        module_id: &ModuleId,
+        ty_args: &[TypeTag],
+        move_storage: &impl MoveStorage,
+    ) -> VMResult<(Vec<TypeTag>, Vec<TypeTag>)> {
+        let mut data_store = TransactionDataCache::new(move_storage, self);
```

Comment:
It would be nice to give clients a way to retain this cache across multiple calls to `get_function_signature` and `get_type_layout`... I could imagine this being important for perf in the dynamic analysis.

Reply:
Counterintuitively, since the loader owns the code cache, the results are indeed cached here. The (dummy) transaction data cache here is merely acting as a passthrough layer to allow the Loader (aka code cache) to access the storage. 

Again, this is because our current ownership model is somewhat screwed up. See my comment below for the fix.